x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/AutoscalingTestCase.java

```diff
@@ -186,7 +186,9 @@ public static AutoscalingMetadata randomAutoscalingMetadataOfPolicyCount(final i
     }
 
     public static SortedSet<String> randomRoles() {
-        return randomSubsetOf(DiscoveryNode.getPossibleRoleNames()).stream().collect(Sets.toUnmodifiableSortedSet());
+        return randomSubsetOf(randomIntBetween(1, DiscoveryNode.getPossibleRoleNames().size() - 1), DiscoveryNode.getPossibleRoleNames())
```

Comment:
I think we can/should leave this out. `randomRoles` is used in a number of tests and the empty set of roles is legal for many of those. The other change should be enough to fix this I think?

Reply:
yeah, this works, I ran it locally 1000s of times and it never failed again.